ls1intum/Artemis
# [Bugfix] Fix showing buildlogs for Swift and Python on Bamboo and Jenkins

### Checklist

### Motivation and Context


Build logs on Bamboo were not shown for Swift and Python programming exercises on Jenkins and Bamboo.

### Description

This PR enables to show the build logs for Swift and Python programming exercises on Jenkins and Bamboo.
Fetching build logs in Bamboo differed from fetching in Jenkins, as they were blacklisted in Bamboo and whitelisted in Jenkins. This is solved by providing filter functions ("isWarningLog", "isErrorLog", etc.) in the build log service which is blacklisting build logs for both CI systems.

Furhermore the Jenkins Pipeline and Bamboo Build Scripts for Python were adapted to be able to show build logs at all. The python code is now compiled at first and `pytest` gets executed only on success.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Swift or a Python Exercise
4. Open the Online Editor "Edit in Editor"
5. Introduce a compilation error (e.g. remove a '{')
6. Verify that the build logs are shown

Note: This PR needs to be testes on ts1/3 (Bamboo) and ts2 (Jenkins)

### Test Coverage

No new functionality was added so the test cases stay the same.

### Screenshots



Python:
Buildlog in Detail View:
![image](https://user-images.githubusercontent.com/33484318/104192387-0a645900-541f-11eb-9828-6b002a7b51db.png)

Build Output:
![image](https://user-images.githubusercontent.com/33484318/104192413-14865780-541f-11eb-8d81-d3488e887282.png)

Swift:
Buildlog in Detail View:
![image](https://user-images.githubusercontent.com/33484318/104193482-74c9c900-5420-11eb-9bcc-78fae5cf5984.png)

SCA Feedback shows in Detail View, despite build error (with SCA enabled):
![image](https://user-images.githubusercontent.com/33484318/104193663-a773c180-5420-11eb-93ac-44db81455763.png)

Build Output:
![image](https://user-images.githubusercontent.com/33484318/104192851-b5751280-541f-11eb-96b5-27cae4d4057b.png)
